docs/api.md

```diff
@@ -94,18 +94,24 @@ onShouldRetry: function (err, retryAttempt, options) {
     console.log("Error", err)
     console.log("Request", err.originalRequest)
     console.log("Response", err.originalResponse)
-    
+
     var status = err.originalResponse ? err.originalResponse.getStatus() : 0
     // Do not retry if the status is a 403.
     if (status === 403) {
       return false
     }
-    
+
     // For any other status code, we retry.
     return true
 }
 ```
 
+#### onUploadUrlAvailable
+
+*Default value:* `null`
+
+An optional function called once the upload URL is retrieved from the `Location` header of the initial creation POST's response. At this point, `upload.url` is guaranteed to be accessible and valid.
```

Comment:
The callback is invoked in two cases: 1) A new upload has been created using the POST method, or 2) an upload URL was confirmed using a HEAD request. In addition, the callback might be invoked multiple times when multiple HEAD requests occur in a single upload due to connection interruptions.

I think these facts should be added to the documentation as well.

Reply:
Sorry for the long delay! I have modified the documentation (as well as my understanding!) accordingly.